internal/storage/v2/grpc/tracereader.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package grpc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"iter"
+
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"google.golang.org/grpc"
+
+	"github.com/jaegertracing/jaeger/internal/storage/v2/api/tracestore"
+	"github.com/jaegertracing/jaeger/proto-gen/storage/v2"
+)
+
+var _ tracestore.Reader = (*TraceReader)(nil)
+
+type TraceReader struct {
+	client storage.TraceReaderClient
+}
+
+func NewTraceReader(tracedConn *grpc.ClientConn) *TraceReader {
+	return &TraceReader{
+		client: storage.NewTraceReaderClient(tracedConn),
+	}
+}
+
+func (t *TraceReader) GetTraces(
```

Comment:
I would recommend starting with a simpler function like GetServices, because the unit tests should run against the corresponding GRPC server (which will be used in remote-storage).

Reply:
@yurishkuro Are you suggesting that we start up a real server for the unit tests? The v1 unit tests were just mocking out the underlying clients.

Reply:
Yes I would suggest that. We often do that in unit tests. The alternative is to do this in e2e tests, but the coverage there is not as good (e2e mostly test happy path)